### PR TITLE
Added cucumber tests for search by name or address

### PR DESCRIPTION
**Pull request by AlexaVB**
Cucumber tests for this page https://courttribunalfinder.service.gov.uk/search/address

### CODE REVIEW

File: cucumber/features/step_definitions/name_or_address_steps.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  name_or_address_page.load_page
+end
+
+Then(/^I should see the search by name or address header$/) do
+  expect(name_or_address_page.page_header.text)
+    .to eq 'Search by name or address'
+end
+
+Then(/^I should see the hint for the form$/) do
+  expect(name_or_address_page.form_block.form_hint.count).to eq 2
```
Comment (richerry) [inline cucumber/features/step_definitions/name_or_address_steps.rb]: Should the number of forms be a separate scenario?

File: cucumber/features/step_definitions/name_or_address_steps.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  expect(name_or_address_page.form_block.form_hint[0].text)
+    .to eq 'Enter one of the following'
+end
+
+When(/^I search for '(.*?)'$/) do |arg1|
```
Comment (richerry) [inline cucumber/features/step_definitions/name_or_address_steps.rb]: Might be good to set a meaningful arg name if possible.

File: cucumber/features/step_definitions/name_or_address_steps.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+end
+
+Then(/^I should see the result is '(.*?)'$/) do |arg1|
+  expect(search_results_page.search_results.number_of_results.text).to eq '1'
+  expect(search_results_page.search_results.court_results.header[0].text)
+    .to eq arg1
+end
+
+Then(/^I should see the top result is Liverpool Civil and Family Court$/) do
+  expect(search_results_page.search_results.number_of_results.text).to eq '10'
```
Comment (richerry) [inline cucumber/features/step_definitions/name_or_address_steps.rb]: Are we testing that there's enough data in, or that pagination is working? Should this line perhaps be in a different scenario?

File: cucumber/features/step_definitions/name_or_address_steps.rb (hunk 1 above)
Comment (AlexaVB) [inline cucumber/features/step_definitions/name_or_address_steps.rb]: Its just checking both parts of the hint are there, its split in two, inspect the element and you'll see :)

File: cucumber/features/step_definitions/name_or_address_steps.rb (hunk 3 above)
Comment (AlexaVB) [inline cucumber/features/step_definitions/name_or_address_steps.rb]: A good scenario but not on this page, should be on search results page

Comment (AlexaVB) [inline cucumber/features/step_definitions/name_or_address_steps.rb]: I don't see pagination on the public site when looking at production.

File: cucumber/features/step_definitions/name_or_address_steps.rb [hunk 4]
```diff
@@ -13,15 +13,15 @@
     .to eq 'Enter one of the following'
 end
 
-When(/^I search for '(.*?)'$/) do |arg1|
-  name_or_address_page.form_block.address.set(arg1)
+When(/^I search for '(.*?)'$/) do |string|
```
Comment (richerry) [inline cucumber/features/step_definitions/name_or_address_steps.rb]: By meaningful arg name I meant something like 'search_term' so the code in the step and any exceptions we encounter while using it, read better.

Comment (AlexaVB) [inline cucumber/features/step_definitions/name_or_address_steps.rb]: Okay, will change this in the next push

**Pull request merged**